In [1]:
import pandas as pd

In [2]:
from google.colab import files
uploaded=files.upload()

Saving retail_store_sales.csv to retail_store_sales.csv


In [34]:
df.head()

,Transaction ID,Customer ID,Category,Item,Price Per Unit,Quantity,Total Spent,Payment Method,Location,Transaction Date,Discount Applied
0,TXN_6867343,CUST_09,Patisserie,Item_10_PAT,18.5,10.0,185.0,Digital Wallet,Online,2024-04-08,True
1,TXN_3731986,CUST_22,Milk Products,Item_17_MILK,29.0,9.0,261.0,Digital Wallet,Online,2023-07-23,True
2,TXN_9303719,CUST_02,Butchers,Item_12_BUT,21.5,2.0,43.0,Credit Card,Online,2022-10-05,False
3,TXN_9458126,CUST_06,Beverages,Item_16_BEV,27.5,9.0,247.5,Credit Card,Online,2022-05-07,True
4,TXN_4575373,CUST_05,Food,Item_6_FOOD,12.5,7.0,87.5,Digital Wallet,Online,2022-10-02,False


In [3]:
df=pd.read_csv('retail_store_sales.csv')
df.head()

,Transaction ID,Customer ID,Category,Item,Price Per Unit,Quantity,Total Spent,Payment Method,Location,Transaction Date,Discount Applied
0,TXN_6867343,CUST_09,Patisserie,Item_10_PAT,18.5,10.0,185.0,Digital Wallet,Online,2024-04-08,True
1,TXN_3731986,CUST_22,Milk Products,Item_17_MILK,29.0,9.0,261.0,Digital Wallet,Online,2023-07-23,True
2,TXN_9303719,CUST_02,Butchers,Item_12_BUT,21.5,2.0,43.0,Credit Card,Online,2022-10-05,False
3,TXN_9458126,CUST_06,Beverages,Item_16_BEV,27.5,9.0,247.5,Credit Card,Online,2022-05-07,NaN
4,TXN_4575373,CUST_05,Food,Item_6_FOOD,12.5,7.0,87.5,Digital Wallet,Online,2022-10-02,False


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12575 entries, 0 to 12574
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Transaction ID    12575 non-null  object 
 1   Customer ID       12575 non-null  object 
 2   Category          12575 non-null  object 
 3   Item              11362 non-null  object 
 4   Price Per Unit    11966 non-null  float64
 5   Quantity          11971 non-null  float64
 6   Total Spent       11971 non-null  float64
 7   Payment Method    12575 non-null  object 
 8   Location          12575 non-null  object 
 9   Transaction Date  12575 non-null  object 
 10  Discount Applied  8376 non-null   object 
dtypes: float64(3), object(8)
memory usage: 1.1+ MB


In [5]:
df.describe()

,Price Per Unit,Quantity,Total Spent
count,11966.000000,11971.000000,11971.000000
mean,23.365912,5.536380,129.652577
std,10.743519,2.857883,94.750697
min,5.000000,1.000000,5.000000
25%,14.000000,3.000000,51.000000
50%,23.000000,6.000000,108.500000
75%,33.500000,8.000000,192.000000
max,41.000000,10.000000,410.000000


In [6]:
df.isnull().sum()

,0
Transaction ID,0
Customer ID,0
Category,0
Item,1213
Price Per Unit,609
Quantity,604
Total Spent,604
Payment Method,0
Location,0
Transaction Date,0


In [9]:
df['Item']=df.groupby('Category')['Item'].transform(lambda x:x.fillna(x.mode()[0]))

In [10]:
df['Price Per Unit']=df['Price Per Unit'].fillna(df['Price Per Unit'].median())

In [11]:
df['Quantity']=df['Quantity'].fillna(df['Quantity'].fillna(df['Quantity'].median()))

In [18]:
df['Total Spent']=df['Total Spent'].fillna(df['Total Spent'].median())

In [22]:
df['Discount Applied']=df['Discount Applied'].fillna(df['Discount Applied'].mode()[0])

/tmp/ipykernel_899/245980323.py:1: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Discount Applied']=df['Discount Applied'].fillna(df['Discount Applied'].mode()[0])


In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12575 entries, 0 to 12574
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Transaction ID    12575 non-null  object 
 1   Customer ID       12575 non-null  object 
 2   Category          12575 non-null  object 
 3   Item              12575 non-null  object 
 4   Price Per Unit    12575 non-null  float64
 5   Quantity          12575 non-null  float64
 6   Total Spent       12575 non-null  float64
 7   Payment Method    12575 non-null  object 
 8   Location          12575 non-null  object 
 9   Transaction Date  12575 non-null  object 
 10  Discount Applied  12575 non-null  bool   
dtypes: bool(1), float64(3), object(7)
memory usage: 994.8+ KB


In [24]:
df.isnull().sum()

,0
Transaction ID,0
Customer ID,0
Category,0
Item,0
Price Per Unit,0
Quantity,0
Total Spent,0
Payment Method,0
Location,0
Transaction Date,0


In [27]:
high_sales = df[df['Total Spent'] > 300]

high_sales.head()

,Transaction ID,Customer ID,Category,Item,Price Per Unit,Quantity,Total Spent,Payment Method,Location,Transaction Date,Discount Applied
27,TXN_1599706,CUST_14,Furniture,Item_25_FUR,41.0,10.0,410.0,Credit Card,In-store,2024-03-24,True
33,TXN_1494700,CUST_12,Electric household essentials,Item_21_EHE,35.0,9.0,315.0,Credit Card,In-store,2024-03-07,True
50,TXN_5379241,CUST_05,Butchers,Item_20_BUT,33.5,10.0,335.0,Credit Card,In-store,2023-04-12,False
60,TXN_6820452,CUST_10,Computers and electric accessories,Item_24_CEA,39.5,8.0,316.0,Credit Card,In-store,2023-04-18,False
64,TXN_6952276,CUST_01,Food,Item_22_FOOD,36.5,10.0,365.0,Credit Card,Online,2024-08-02,True


In [28]:
df.sort_values('Total Spent', ascending=False).head(10)

,Transaction ID,Customer ID,Category,Item,Price Per Unit,Quantity,Total Spent,Payment Method,Location,Transaction Date,Discount Applied
8611,TXN_3342443,CUST_08,Butchers,Item_25_BUT,41.0,10.0,410.0,Credit Card,In-store,2023-10-03,True
1060,TXN_3710081,CUST_25,Furniture,Item_25_FUR,41.0,10.0,410.0,Digital Wallet,In-store,2023-08-12,True
8596,TXN_2288041,CUST_24,Food,Item_14_FOOD,23.0,10.0,410.0,Digital Wallet,In-store,2023-09-02,False
1088,TXN_1273334,CUST_22,Beverages,Item_25_BEV,41.0,10.0,410.0,Digital Wallet,In-store,2022-11-04,True
11610,TXN_6812555,CUST_02,Beverages,Item_25_BEV,41.0,10.0,410.0,Cash,In-store,2024-01-24,True
1983,TXN_6924479,CUST_21,Patisserie,Item_25_PAT,41.0,10.0,410.0,Credit Card,Online,2022-06-11,True
11608,TXN_7169895,CUST_07,Food,Item_25_FOOD,41.0,10.0,410.0,Digital Wallet,In-store,2022-05-11,True
133,TXN_2953434,CUST_25,Furniture,Item_25_FUR,41.0,10.0,410.0,Credit Card,In-store,2023-08-10,True
6611,TXN_7056530,CUST_09,Food,Item_25_FOOD,41.0,10.0,410.0,Credit Card,Online,2022-08-15,True
11662,TXN_5870737,CUST_17,Computers and electric accessories,Item_25_CEA,41.0,10.0,410.0,Credit Card,Online,2024-12-24,False


In [29]:
category_sales = df.groupby('Category')['Total Spent'].sum()

category_sales

,Total Spent
Category,
Beverages,204751.0
Butchers,215930.0
Computers and electric accessories,199481.0
Electric household essentials,211951.0
Food,203600.5
Furniture,202471.0
Milk Products,187815.5
Patisserie,191605.0


In [31]:
df.head(3)

,Transaction ID,Customer ID,Category,Item,Price Per Unit,Quantity,Total Spent,Payment Method,Location,Transaction Date,Discount Applied
0,TXN_6867343,CUST_09,Patisserie,Item_10_PAT,18.5,10.0,185.0,Digital Wallet,Online,2024-04-08,True
1,TXN_3731986,CUST_22,Milk Products,Item_17_MILK,29.0,9.0,261.0,Digital Wallet,Online,2023-07-23,True
2,TXN_9303719,CUST_02,Butchers,Item_12_BUT,21.5,2.0,43.0,Credit Card,Online,2022-10-05,False


In [33]:
pivot_table = pd.pivot_table(
    df,
    values='Total Spent',
    index='Payment Method',
    columns='Category',
    aggfunc='sum'
)

pivot_table

Category,Beverages,Butchers,Computers and electric accessories,Electric household essentials,Food,Furniture,Milk Products,Patisserie
Payment Method,,,,,,,,
Cash,73070.5,75350.5,70884.0,70289.5,73182.0,69518.0,63137.0,64738.0
Credit Card,63697.5,75545.0,66494.5,62792.5,66144.5,67383.0,63209.0,62865.0
Digital Wallet,67983.0,65034.5,62102.5,78869.0,64274.0,65570.0,61469.5,64002.0


**Key Insights**
1)Cash generated the highest sales for Butchers, with total sales of 75,350.5.

2)Digital Wallet generated the highest sales for Electric household essentials, with 78,869.0.

3)Credit Card generated the highest sales for Butchers, with 75,545.0 — slightly higher than Cash.

4)Milk Products had the lowest sales across all payment methods, with values ranging from 61,469.5 to 63,209.0.

5)The highest individual value in the entire Pivot Table is 78,869.0, for Electric household essentials using Digital Wallet.

In [44]:
final_dataset=df.to_csv("final_dataset.csv",index=False)

In [46]:
from google.colab import files

files.download("final_dataset.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>